protocol/contracts/beanstalk/silo/SiloFacet/SiloExit.sol

```diff
@@ -0,0 +1,333 @@
+     * computational complexity.
+     */
+    function balanceOfStalk(address account) public view returns (uint256) {
+        return s.a[account].s.stalk.add(balanceOfEarnedStalk(account));
+    }
+
+    /**
+     * @notice Returns the balance of Roots for `account`.
+     * @dev Roots within Beanstalk are entirely separate from the 
+     * [ROOT ERC-20 token](https://roottoken.org/).
+     * 
+     * Roots represent proportional ownership of Stalk:
+     *  `balanceOfStalk / totalStalk = balanceOfRoots / totalRoots`
+     * 
+     * Roots are used to calculate Earned Bean, Earned Stalk and Plantable Seed
+     * balances.
+     *
+     * FIXME(doc): how do Roots relate to Raining?
+     */
+    function balanceOfRoots(address account) public view returns (uint256) {
+        return s.a[account].roots;
+    }
+
+    /**
+     * @notice Returns the balance of Grown Stalk for `account`. Grown Stalk is 
+     * earned each Season from Seeds and must be Mown via `SiloFacet-mow` to 
+     * apply it to a user's balance.
+     * @dev The balance of Grown Stalk for an account can be calculated as:
+     *
+     * ```
+     * elapsedSeasons = currentSeason - lastUpdatedSeason
+     * grownStalk = balanceOfSeeds * elapsedSeasons
+     * ```
+     */
+    function balanceOfGrownStalk(address account)
+        public
+        view
+        returns (uint256)
+    {
+        return
+            LibSilo.stalkReward(
+                s.a[account].s.seeds,
+                _season() - lastUpdate(account)
+            );
+    }
+    
+    /**
+     * @notice Returns the balance of Earned Beans for `account`. Earned Beans
+     * are the Beans distributed to Stalkholders during {Sun-rewardToSilo}.
+     */
+    function balanceOfEarnedBeans(address account)
+        public
+        view
+        returns (uint256 beans)
+    {
+        beans = _balanceOfEarnedBeans(account, s.a[account].s.stalk);
+    }
+
+    /**
+     * @dev Internal function to compute `account` balance of Earned Beans.
+     *
+     * The number of Earned Beans is equal to the difference between: 
+     *  - the "expected" Stalk balance, determined from the account balance of 
+     *    Roots. 
+     *  - the "account" Stalk balance, stored in account storage.
+     * divided by the number of Stalk per Bean.
+     * The earned beans from the latest season 
+     */
+    function _balanceOfEarnedBeans(address account, uint256 accountStalk) 
+        internal
+        view
+        returns (uint256 beans) {
+        // There will be no Roots before the first Deposit is made.
+        if (s.s.roots == 0) return 0;
+
+        uint256 stalk;
+        if(block.number - s.season.sunriseBlock <= EARNED_BEAN_VESTING_BLOCKS){
+            stalk = s.s.stalk.sub(s.newEarnedStalk).mulDiv(
+                s.a[account].roots.add(s.a[account].deltaRoots), // add the delta roots of the user
+                s.s.roots.add(s.newEarnedRoots), // add delta of global roots 
```

Comment:
s.newEarnedBeans is set when calling mow, which may not be called before checking of balance here. should we trigger an update before this addition?